From 7b6ce05e789ca90d6116e8b4dd94cafed2b52fdb Mon Sep 17 00:00:00 2001
From: Nancy Chen <Nancy.Chen@freescale.com>
Date: Tue, 3 Jun 2008 10:11:17 -0500
Subject: [PATCH] ENGR00077428 PMIC register setting not correct

PMIC registers are not setup correctly by pmic_init_voltage()

Signed-off-by: Nancy Chen <Nancy.Chen@freescale.com>
---
 arch/arm/mach-mx3/dvfs_v2.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx3/dvfs_v2.c b/arch/arm/mach-mx3/dvfs_v2.c
index 26c8043..7f525ca 100644
--- a/arch/arm/mach-mx3/dvfs_v2.c
+++ b/arch/arm/mach-mx3/dvfs_v2.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2007-2008 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -331,8 +331,10 @@ void pmic_voltage_init(void)
 	pmic_write_reg(REG_INTERRUPT_MASK_1, 0, (1 << 11));
 
 	/* Set mc13783 DVS speed 25mV each 4us */
-	pmic_write_reg(REG_SWITCHERS_4, 1, (1 << 6));
-	pmic_write_reg(REG_SWITCHERS_4, 0, (1 << 7));
+	pmic_write_reg(REG_SWITCHERS_4, 1 << 6, (3 << 6));
+
+	/* Set mc13783 SW1A operating mode in normal mode */
+	pmic_write_reg(REG_SWITCHERS_4, (1 << 0), (3 << 0));
 
 	volt.sw1a = SW1A_1_625V;
 	pmic_power_regulator_set_voltage(SW_SW1A, volt);
-- 
1.5.4.4

